golang/cosmos/app/upgrade.go

```diff
@@ -72,6 +75,47 @@ func isFirstTimeUpgradeOfThisVersion(app *GaiaApp, ctx sdk.Context) bool {
 	return true
 }
 
+func replaceElectorateCoreProposalStep(upgradeName string) (vm.CoreProposalStep, error) {
+	t := template.Must(template.New("").Parse(`{
+		"module": "@agoric/builders/scripts/inter-protocol/replace-electorate-core.js",
+    	"entrypoint": "defaultProposalBuilder",
+		"args": [{
+			"variant": {{.variantJson}}
+		}]
+	}`))
+
+	var variant string
+
+	switch validUpgradeName(upgradeName) {
+	case "UNRELEASED_A3P_INTEGRATION":
+		variant = "A3P_INTEGRATION"
+	case "UNRELEASED_main":
+		variant = "MAINNET"
+	case "UNRELEASED_devnet":
+		variant = "DEVNET"
+	// Noupgrade for this version.
+	case "UNRELEASED_BASIC":
+	}
+	variantJson, err := json.Marshal(variant)
+	if err != nil {
+		return nil, err
+	}
+	var result strings.Builder
+	err = t.Execute(&result, map[string]any{
+		"variantJson": string(variantJson),
+	})
+	if err != nil {
+		return nil, err
+	}
+	jsonStr := result.String()
+	jsonBz := []byte(jsonStr)
+	if !json.Valid(jsonBz) {
+		return nil, fmt.Errorf("invalid JSON: %s", jsonStr)
+	}
+	proposal := vm.ArbitraryCoreProposal{Json: jsonBz}
+	return vm.CoreProposalStepForModules(proposal), nil
```

Comment:
I'm going to need to do all of this for the priceFeed upgrade. Can it be defined so it can be shared rather than requiring me to copy it?

Reply:
Thanks!  I build on top of this for #10317